include/zephyr/bluetooth/mesh/msg.h

```diff
@@ -72,6 +72,27 @@ struct bt_mesh_model;
 #define BT_MESH_MODEL_BUF_DEFINE(_buf, _op, _payload_len)                      \
 	NET_BUF_SIMPLE_DEFINE(_buf, BT_MESH_MODEL_BUF_LEN(_op, (_payload_len)))
 
+/**
+ *  @brief Define and initialize a Mesh model message buffer using @ref NET_BUF_SIMPLE_DEFINE.
+ *	   The message buffer will be initialized with the opcode @c _op and list of payload
+ *	   data passed as additional arguments to the macro.
+ *
+ *	   The opcode @c _op must be declared as a macro and shall have `<_op>_RAW`
+ *	   macro pair that defines the raw data of the opcode.
+ *
+ *	   The multiple-octet values can be encoded using macros defined in @ref
+ *	   bt_byteorder.
+ *
+ *  @param _buf         Buffer name.
+ *  @param _op          Opcode of the message.
+ *  @param _payload_len Length of the model message payload.
+ *  @param ...		List of payload data.
+ */
+#define BT_MESH_MODEL_BUF_INIT(_buf, _op, _payload_len, ...)			\
+	BUILD_ASSERT(sizeof((uint8_t []){ __VA_ARGS__ }) <= _payload_len);	\
+	NET_BUF_SIMPLE_DEFINE(_buf, BT_MESH_MODEL_BUF_LEN(_op, _payload_len),	\
+			      _op ## _RAW, ##__VA_ARGS__)
```

Comment:
I'm not sure I'd make the `BT_MESH_MODEL_BUF_INIT()` API require the existence of a define whose name must follow the pattern `OP_*_RAW`. We had to have this kind of required naming scheme with the earlier `_LEN` solution, but I don't think there's really any reason to require it now. Someone could be perfectly happy to define their opcodes using only the new _SIG() or _VND() macros, and then use the `BT_MESH_MODEL_OPCODE()` conversion explicitly in the places where it's needed.

Reply:
So, how todo ?
Change to directly use `OP_*_RAW`?

Reply:
Yes, the `BT_MESH_MODEL_BUF_INIT()` would take as input a value created using the new OP_STD() or OP_VND() macros, which in your current proposal is the *_RAW defines.